src/runtime_src/hip/core/common.h

```diff
@@ -42,32 +52,68 @@ insert_in_map(map& m, value&& v)
   m.add(handle, std::move(v));
   return handle;
 }
+
+template <class T>
+std::string
+get_untangled_type_name(const T& obj)
+{
+  const char* mangled_name = typeid(obj).name();
+#if defined(__GNUC__) || defined(__clang__)
+  int status = 0;
+  std::unique_ptr<char, decltype(&std::free)> demangled_name(
+    abi::__cxa_demangle(mangled_name, nullptr, nullptr, &status), &std::free
+  );
+  return (status == 0) ? demangled_name.get() : mangled_name;
+#elif defined(_MSC_VER)
+  // MSVC's typeid().name() returns a readable name by default
+  return mangled_name;
+#else
+  return mangled_name;
+#endif
+}
 } // xrt::core::hip
 
 namespace {
+
 template<typename F> hipError_t
 handle_hip_func_error(const char* func_name, hipError_t default_err, F && f)
 {
+  xrt::core::hip::error::instance().reset_local_errors();
   try {
     std::forward<F>(f)();
     return hipSuccess;
   }
+  catch (const xrt::core::hip::hip_exception &ex) {
```

Comment:
```suggestion
  catch (const xrt::core::hip::hip_exception& ex) {
```
Nit, most commonly used.